go/externals/services.go

```diff
@@ -4,37 +4,130 @@
 package externals
 
 import (
+	"context"
 	"strings"
+	"sync"
 
 	libkb "github.com/keybase/client/go/libkb"
 )
 
-type externalServicesCollection map[string]libkb.ServiceType
+func getStaticProofServices() []libkb.ServiceType {
+	return []libkb.ServiceType{
+		DNSServiceType{},
+		FacebookServiceType{},
+		GithubServiceType{},
+		HackerNewsServiceType{},
+		RedditServiceType{},
+		RooterServiceType{},
+		TwitterServiceType{},
+		WebServiceType{},
+	}
+}
+
+// staticProofServies are only used for testing or for basic assertion
+// validation
+type staticProofServices struct {
+	collection map[string]libkb.ServiceType
+}
+
+func NewStaticProofServices() libkb.ExternalServicesCollector {
+	staticServices := getStaticProofServices()
+	p := staticProofServices{
+		collection: make(map[string]libkb.ServiceType),
+	}
+	for _, st := range staticServices {
+		p.register(st)
+	}
+	return &p
+}
+
+func (p *staticProofServices) register(st libkb.ServiceType) {
+	for _, k := range st.AllStringKeys() {
+		p.collection[k] = st
```

Comment:
do you need `useDevelProofCheckers || !v.IsDevelOnly()` here too?

Reply:
ah good catch